crowbar_framework/lib/crowbar/checks/network.rb

```diff
@@ -0,0 +1,147 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "resolv"
+require "ipaddr"
+require "socket"
+require "logger"
+
+module Crowbar
+  module Checks
+    class Network
+      attr_accessor :logger
+
+      class << self
+        def ip_addrs(fqdn, version = :ipv4)
+          ip_addrs = []
+
+          Resolv.getaddresses(fqdn).each do |address|
+            ip_addr = IPAddr.new(address)
+            if version == :ipv6
+              ip_addrs.push address if ip_addr.ipv6?
+            else
+              ip_addrs.push address if ip_addr.ipv4?
+            end
+          end
+
+          ip_addrs
+        end
+      end
+
+      def initialize
+      end
+
+      def fqdn_detected?
+        if hostname.blank? || domain.blank? || fqdn.blank?
+          return false
+        end
+
+        true
+      end
+
+      def ip_resolved?
+        if ipv4_addrs.empty? && ipv6_addrs.empty?
+          return false
+        end
+
+        true
+      end
+
+      def loopback_resolved?
+        if ipv4_addrs.find { |e| /^127/ =~ e } || ipv6_addrs.find { |e| /^::[0-9]$/ =~ e }
+          return false
+        end
+
+        true
+      end
+
+      def ip_configured?
+        ipv4_configured = false
+        ipv6_configured = false
+
+        addr_infos = Socket.ip_address_list
+        addr_infos.each do |addr_info|
+          ipv4_configured = true if ipv4_addrs.include?(addr_info.ip_address)
+          ipv6_configured = true if ipv6_addrs.include?(addr_info.ip_address)
+        end
+
+        unless ipv4_configured || ipv4_addrs.empty?
+          return false
+        end
+        unless ipv6_configured || ipv6_addrs.empty?
+          return false
+        end
+
+        true
+      end
+
+      def network_valid?
+        if File.exist?("/etc/crowbar/network.json")
+          network_json = "/etc/crowbar/network.json"
+        else
+          network_json = Rails.root.join(
+            "..", "chef", "data_bags", "crowbar", "template-network.json").to_s
+        end
+
+        admin_ip = NodeObject.admin_node.ip
+        unless ipv4_addrs.include?(admin_ip)
+          return false
+        end
+        unless system("network-json-validator --admin-ip \"#{admin_ip}\" #{network_json}")
+          return false
+        end
+
+        true
+      end
+
+      def firewall_disabled?
+        if system("LANG=C iptables -n -L | grep -qvE '^$|^Chain [^ ]|^target     prot'")
+          return false
+        end
+
+        true
+      end
+
+      def ping_succeeds?
+        unless system("ping -c 1 #{fqdn}")
+          return false
+        end
+
+        true
+      end
+
+      def hostname
+        @hostname ||= `hostname -s`.strip
+      end
+
+      def domain
+        @domain ||= `hostname -d`.strip
+      end
+
+      def fqdn
+        @fqdn ||= `hostname -f`.strip
+      end
+
+      def ipv4_addrs
+        @ipv4_addrs ||= self.class.ip_addrs(fqdn, "ipv4")
+
+      def ipv6_addrs
```

Comment:
Lint/NestedMethodDefinition: Method definitions must not be nested. Use `lambda` instead. (https://github.com/bbatsov/ruby-style-guide#no-nested-methods)<br>Style/IndentationConsistency: Inconsistent indentation detected.